### ISSUE TITLE
rsync always uses `ssh -4` as default remote shell, even when --ipv6 is specified

### ISSUE THREAD

**Issue by osnyx** [RsyncProject/rsync#648]
I'd like to rsync via ssh to a host only reachable via IPv6. This fails at the SSH connection stage.

It turns out that ssh is always invoked to use IPv4 only. This happens even when specifying `--ipv6` to rsync.

```
/nix/store/f2w6bm4bpgdl3ih1ch6m3dacr3l8fawv-rsync-3.3.0/bin/rsync --ipv6 -rlptzvvv --progress --delete --exclude=.git --exclude=.vscode --exclude=result --exclude=channels/ --exclude=.mypy_cache . testhost:fc-nixos
opening connection using: ssh -4 testhost rsync --server -vvvltprze.iLsfxCIvu --delete . fc-nixos  (9 args)
ssh: connect to host testhost port 22: Network is unreachable
rsync: connection unexpectedly closed (0 bytes received so far) [sender]
rsync error: unexplained error (code 255) at io.c(231) [sender=3.3.0]
[sender] _exit_cleanup(code=12, file=io.c, line=231): about to call exit(255)
```

I can barely imagine that I'm the first one to notice such a bug, so am I possibly doing something wrong here?

Version: rsync-3.3.0
Platform: macOS, arm64, package installed from nixpkgs

**Comment by schmittlauch** [comment]
Interestingly, under Linux (amd64), the nixpkgs package correctly uses `ssh -6`. And an rsync-3.3.0 installed via homebrew on macOS works correctly.

So this might be indeed a Nix packaging issue specific to macOS. I'll keep you updated on this.

**Comment by osnyx** [comment]
Turns out this is fixed by #558, the fix just isn't included in a release yet.
